src/components/menu/__internal__/submenu/submenu.component.js

```diff
@@ -216,53 +275,36 @@ const Submenu = React.forwardRef(
             setTimeout(() => closeSubmenu());
           }
 
-          if (href && index === undefined) {
+          if (href && submenuFocusIndex === undefined) {
             if (
               Events.isEnterKey(event) ||
               (Events.isTabKey(event) && Events.isShiftKey(event))
             ) {
               closeSubmenu();
               return;
             }
-
-            if (
-              Events.isSpaceKey(event) ||
-              Events.isDownKey(event) ||
-              Events.isUpKey(event) ||
-              Events.isTabKey(event)
-            ) {
-              nextIndex = 0;
-            }
           }
 
-          // Defensive check in case an unhandled key event from a child component
-          // has bubbled up
-          if (!nextIndex && nextIndex !== 0) return;
-
-          // Check that next index contains a MenuItem
-          // If not, call handleKeyDown again
```

Comment:
I seem to remember that this block of code was required to manage the keyboard navigation if the submenu block had a `Search` component. GIF of how it behaves on this PR.

![submenu-with-search-bug](https://user-images.githubusercontent.com/56251247/214321404-2f26ce08-8b50-4f82-be92-105e98563301.gif)

Reply:
good point I'll need to include the close in the query string

Reply:
I've re-tested this in the latest commits and it works better than before now 😆 
I'll resolve this comment.